prow/pjutil/health.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package pjutil contains helpers for working with ProwJobs.
+package pjutil
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Health keeps a request multiplexer for health liveness and readiness endpoints
+type Health struct {
+	healthMux *http.ServeMux
+}
+
+// NewHealth creates a new health request multiplexer and starts serving the liveness endpoint
+// on the given port
+func NewHealth(port int) *Health {
```

Comment:
Why do we need to parameterize this?

Reply:
It might be that some consumers of this health endpoint might want to configure a different port than 8081.

Reply:
a) if every instance of this uses 8081 it suggests we don't actually need this argument right now
b) at least use a constant?
c) only tot and deck have health checks?

Reply:
I exacted the port into a constant.

Reply:
c) also `hook` has a health check but on the root path

I integrated this health check also into the hook service, but still keep the old one for compatibility. 

Also added the liveness and readiness probes to hook deployment.